### ISSUE TITLE
Not working

### ISSUE THREAD

**Issue by Llspud82** [zecoj/kodi-tvvn#9]
Hi,

Just want to let u know that this addon stopped working a few days ago. Is it being fixed Also? Also will it be ported over to the recent version of Kodi. Thanks

**Comment by sondo01** [comment]
> Hi,
> 
> Just want to let u know that this addon stopped working a few days ago. Is it being fixed Also? Also will it be ported over to the recent version of Kodi. Thanks

I found a quick hack. In tvvn.py, in play_link function, before d_progress.close(), add the following lines of code:
`    stringA = opener.open(full_url).read().decode('utf-8')`
`    stringB = "chunklist_"`
`    stringC = ".m3u8"`
`    chunk_id = re.search(stringB+"(.*?)"+re.escape(stringC),stringA).group(1)`
`    pre_urlA = full_url.split("/")`
`    del pre_urlA[-1]`
`    full_url_BC = "/".join(pre_urlA) +"/"+ stringB + chunk_id + stringC`
And replace 
`    xbmc.Player().play(full_url)`
with 
`    xbmc.Player().play(full_url_BC)`

Basically, the extra lines search for the real URL of each chunklist.
Tested on Kodi 19.1 with http://www.tvnet.gov.vn/ source.

**Comment by Llspud82** [comment]
> > Hi,
> > Just want to let u know that this addon stopped working a few days ago. Is it being fixed Also? Also will it be ported over to the recent version of Kodi. Thanks
> 
> I found a quick hack. In tvvn.py, in play_link function, before d_progress.close(), add the following lines of code: ` stringA = opener.open(full_url).read().decode('utf-8')` ` stringB = "chunklist_"` ` stringC = ".m3u8"` ` chunk_id = re.search(stringB+"(.*?)"+re.escape(stringC),stringA).group(1)` ` pre_urlA = full_url.split("/")` ` del pre_urlA[-1]` ` full_url_BC = "/".join(pre_urlA) +"/"+ stringB + chunk_id + stringC` And replace ` xbmc.Player().play(full_url)` with ` xbmc.Player().play(full_url_BC)`
> 
> Basically, the extra lines search for the real URL of each chunklist. Tested on Kodi 19.1 with http://www.tvnet.gov.vn/ source.

Hi,

Thanks for getting back to me. Sorry I only just saw the reply. I can do that however how do I edit the file within Kodi? I'm brand new to this. Is there a online guide for this? Also I have kodi installed in my tv. Thanks

**Comment by sondo01** [comment]
> Hi,
> 
> Thanks for getting back to me. Sorry I only just saw the reply. I can do that however how do I edit the file within Kodi? I'm brand new to this. Is there a online guide for this? Also I have kodi installed in my tv. Thanks

Hi, 
I committed the patch but don't know if the authors will accept it.
The easiest way for you to do is:

1. - download this project as zip file. 
2. - Unzip it, edit the tvvn.py file then zip it again 
3. - Finally manually install the "new" version from zip file.

For the step 2, search for the  line
`#m3u8 url from tvnet`
replace the code below it  to the line 
`#m3u8 url from vtvgo`

by 

```
elif data['channels'][chn]['src']['playpath'] == "m3u8_tvnet":
        url = 'http://au.tvnet.gov.vn/kenh-truyen-hinh/'+data['channels'][chn]['src']['page_id']
        stringA = opener.open(url).read().decode('utf-8')
        stringB = 'let ownURL = "'
        stringC = '"'
        url = re.search(stringB+"(.*?)"+re.escape(stringC),stringA).group(1)
        
        stringA = opener.open(url).read().decode('utf-8')
        stringB = '"url": "'
        stringC = '"'
        full_url_BC = re.search(stringB+"(.*?)"+re.escape(stringC),stringA).group(1)
        stringA = opener.open(full_url_BC).read().decode('utf-8')
        stringB = "chunklist_"
        stringC = ".m3u8"
        chunk_id = re.search(stringB+"(.*?)"+re.escape(stringC),stringA).group(1)
        pre_urlA = full_url_BC.split("/")
        del pre_urlA[-1]
        full_url = "/".join(pre_urlA) +"/"+ stringB + chunk_id + stringC
```

Make sure the lines are indented properly if not Python will not be happy.
Hope it help
